### ISSUE TITLE
more specific details about the execution order

### ISSUE THREAD

**Issue by Bingoyww** [wuyang98/weathergen#2]
This is great work. However, there are some imperfections. We would like to know more specific details about the execution order, such as what programs to run after downloading KITTI-360 and finally executing the training code. Additionally, there are multiple preloaded .pth files in the project, such as diffusion_steps.pth. Is this file the same as the one provided via Baidu Netdisk?

After downloading KITTI-360, what programs should be executed next? For example:
{  
evaluate.py

parser.add_argument("--sample_dir", type=str, default='/opt/data/private/project/points2image/lidm_points')

Where does this file come from?
}

**Comment by wuyang98** [comment]
Hello, after downloading KITTI-360, if you wish to run the training, you can use the command:
accelerate launch --mixed_precision 'fp16' --dynamo_backend 'no' train.py.
After training, you may download the Seeing Through Fog dataset for fine-tuning. Load the pre-trained weights at ​​line 180​​ in train.py, and adjust the training function at ​​line 311​​ accordingly.
To generate point clouds, download the weight file and modify the path at ​​line 80​​ in generate.py, then execute the script.
In evaluate.py, the --sample_dir argument specifies the directory path for storing generated point clouds.

**Comment by Bingoyww** [comment]
Thank you to the author for the reply, it was very helpful. During the training process, it seems that the Seeing Through Fog dataset was used. For example，
` 
# train.py
if weather_flag == 'rain':
    text_features = torch.cat([text_features, text_features[3].unsqueeze(0)], dim=0)
    x_weather = stf_process(weather_flag, x_0)
`

`
#utils/weather.py
if weather_flag == 'rain':
    all_point_path = np.genfromtxt('/./seeingthroughfog/rain.txt', dtype='U', delimiter='\n')
    selected_path = np.random.choice(all_point_path, size=B, replace=False)
`
Another question I have is that I couldn’t find the rain.txt file. Is it generated by filtering the corresponding JSON file names from the seeingthroughfog/labeltool_labels folder? Then, does the code randomly select one of the rainy JSON files?

`
#labeltool_labels/2018-02-08_17-47-02_00100.json
    "weather": {
        "clear": false,
        "dense_fog": false,
        "light_fog": false,
        "rain": true,
        "snow": false
    }
`
So, we can find all the rainy JSON files:
{2018-02-08_17-47-02_00100.json
2018-02-08_17-47-02_00200.json
2018-02-08_17-47-02_00350.json
2018-02-08_17-47-02_00360.json,....}


Is this the right way to do it? And does it align with the author’s perspective?

**Comment by Bingoyww** [comment]
Thank you! Another test regarding the generate.py issue: when inputting two different bins and setting the text to “Rain” with a fixed random seed (seeds=[1]) and batch_size=1, the generated bev.png images turn out to be  very similar. Additionally, the generated images significantly alter the original input structure. Is it possible to configure it to preserve the main original structure?
{
'KITTI-360/data_3d_raw/2013_05_28_drive_0004_sync/velodyne_points/data/0000000018.bin'
and 'KITTI-360/data_3d_raw/2013_05_28_drive_0004_sync/velodyne_points/data/0000000111.bin'

rng=utils.inference.setup_rng([1], device=args.device),

}

**Comment by wuyang98** [comment]
Hello! Thank you for your observation. You are right. This occurs because the fine-tuned model primarily learns ​​external stylized features​​, resulting in nearly identical generated outcomes (with minor variations) when the random seed is fixed. I have verified that the ​​non-fine-tuned model does not exhibit this behavior​​. If you would like to test it, you can download the model via Aliyun (my current institution only permits uploads to Aliyun; the link is valid for 24 hours: https://www.alipan.com/t/xX0RTFoI8b6GsvxHQkYb) and adjust ​​lines 42 and 43​​ in generate.py. Currently, our model cannot generate structurally consistent LiDAR point cloud data, primarily due to the lack of ​​paired training data​​.

**Comment by llddph** [comment]
> Hello! Thank you for your observation. You are right. This occurs because the fine-tuned model primarily learns ​​external stylized features​​, resulting in nearly identical generated outcomes (with minor variations) when the random seed is fixed. I have verified that the ​​non-fine-tuned model does not exhibit this behavior​​. If you would like to test it, you can download the model via Aliyun (my current institution only permits uploads to Aliyun; the link is valid for 24 hours: https://www.alipan.com/t/xX0RTFoI8b6GsvxHQkYb) and adjust ​​lines 42 and 43​​ in generate.py. Currently, our model cannot generate structurally consistent LiDAR point cloud data, primarily due to the lack of ​​paired training data​​.

Thanks for your great work! Could you please resend a valid Aliyun link for the non-fine-tuned model weights?

**Comment by wuyang98** [comment]
> > Hello! Thank you for your observation. You are right. This occurs because the fine-tuned model primarily learns ​​external stylized features​​, resulting in nearly identical generated outcomes (with minor variations) when the random seed is fixed. I have verified that the ​​non-fine-tuned model does not exhibit this behavior​​. If you would like to test it, you can download the model via Aliyun (my current institution only permits uploads to Aliyun; the link is valid for 24 hours: https://www.alipan.com/t/xX0RTFoI8b6GsvxHQkYb) and adjust ​​lines 42 and 43​​ in generate.py. Currently, our model cannot generate structurally consistent LiDAR point cloud data, primarily due to the lack of ​​paired training data​​.
> 
> Thanks for your great work! Could you please resend a valid Aliyun link for the non-fine-tuned model weights?

Hello, thank you. You can try this link: https://www.alipan.com/t/a6Yluh10ZVkbwF6L8kw9

**Comment by llddph** [comment]
Many thanks for sharing the link. I’ve received it and will verify the download shortly.

**Comment by CC-1997** [comment]
I would also like to obtain this set of weights. Could you please send me a copy as well? Thank you very much!

**Comment by Malekithlcx** [comment]
> > > Hello! Thank you for your observation. You are right. This occurs because the fine-tuned model primarily learns ​​external stylized features​​, resulting in nearly identical generated outcomes (with minor variations) when the random seed is fixed. I have verified that the ​​non-fine-tuned model does not exhibit this behavior​​. If you would like to test it, you can download the model via Aliyun (my current institution only permits uploads to Aliyun; the link is valid for 24 hours: https://www.alipan.com/t/xX0RTFoI8b6GsvxHQkYb) and adjust ​​lines 42 and 43​​ in generate.py. Currently, our model cannot generate structurally consistent LiDAR point cloud data, primarily due to the lack of ​​paired training data​​.您好！感谢您的观察。您说得对。出现这种情况是因为微调后的模型主要学习外部的风格化特征，导致在随机种子固定的情况下，生成的结果几乎完全相同（只有细微差别）。我已经验证过，未微调的模型不存在这种现象。如果您想测试一下，可以通过阿里云下载模型（我所在的机构只允许上传到阿里云；链接有效期为 24 小时： https://www.alipan.com/t/xX0RTFoI8b6GsvxHQkYb ），然后修改 generate.py 文件中的第 42 行和第 43 行。目前，我们的模型无法生成结构一致的 LiDAR 点云数据，主要是因为缺乏配对的训练数据。
> > 
> > 
> > Thanks for your great work! Could you please resend a valid Aliyun link for the non-fine-tuned model weights?感谢您的辛勤工作！能否请您重新发送一个有效的阿里云链接，其中包含未经微调的模型权重？
> 
> Hello, thank you. You can try this link: https://www.alipan.com/t/a6Yluh10ZVkbwF6L8kw9您好，谢谢。您可以试试这个链接： https://www.alipan.com/t/a6Yluh10ZVkbwF6L8kw9

您好，能重新发一份吗？
